frontend/src/components/search/__tests__/AdvancedSearch.test.tsx

```diff
@@ -0,0 +1,213 @@
+import '@testing-library/jest-dom';
+import { render, waitFor } from '@testing-library/react'
+import AdvancedSearch from "../AdvancedSearch"
+import { QueryClient, QueryClientProvider } from "react-query";
+import { MemoryRouter, Routes, Route } from "react-router-dom";
+import userEvent from "@testing-library/user-event";
+
+const mockUseNavigate = jest.fn();
+jest.mock('react-router-dom', () => ({
+   ...jest.requireActual('react-router-dom') as any,
+  useNavigate: () => mockUseNavigate,
+}));
+
+const mockGetAllCategories = jest.fn()
+jest.mock('../../../api/categories', () => ({
+    getAllCategories: () => mockGetAllCategories(),
+}))
+
+const queryClient = new QueryClient()
+const user = userEvent.setup()
+
+const renderPage = (
+    queryClient: QueryClient,
+    collection = "example collection" 
+    ) => {
+        return render(
+            <QueryClientProvider client={queryClient}>
+                <MemoryRouter initialEntries={[`/collections/${collection}/artworks`]}>
+                    <Routes>
+                        <Route path="/collections/:collection/artworks" element={<AdvancedSearch collectionName={collection} />}/>
+                    </Routes>  
+                </MemoryRouter>
+            </QueryClientProvider>
+        );
+};
+
+const fetchedCategories = {
+    categories: [
+        "Tytuł",
+        "Tytuł.Podtytuł",
+        "Artyści",
+        "Rok"
+    ]
+}
+
+describe("AdvancedSearch tests", () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+        queryClient.clear();
+    });
+
+    test("component renders correctly", async () => {        
+        const {getByTestId, queryByTestId} = renderPage(queryClient)
+
+        expect(getByTestId("loading-advanced-search-container")).toBeInTheDocument()
+        expect(queryByTestId("advancedSearchComponent")).not.toBeInTheDocument()
```

Comment:
Testy na rendering pisze się troszkę inaczej
```suggestion
        const { container } = renderPage(queryClient)

        expect(container).toMatchSnapshot()
```

Reply:
Chodzi mi o to, że testy są też po to, żeby ułatwić zrozumienie implementacji innym programistom co usiadą po Tobie do jest apki, więc warto jasno komunikować, co chcesz sprawdzić w tym. Wyrenderować się zawsze wyrenderuje, bardziej warto określić co dla danej sytuacji ma się wyrenderować ;)